tests/adapter/dbt/tests/adapter/materialized_view/on_configuration_change.py

```diff
@@ -0,0 +1,91 @@
+from typing import List
+
+import pytest
+import yaml
+
+from dbt.tests.util import read_file, write_file, relation_from_name
+from dbt.contracts.results import RunStatus
+
+from dbt.tests.adapter.materialized_view.base import (
+    Base,
+    assert_message_in_logs,
+)
+
+
+def get_project_config(project):
+    file_yaml = read_file(project.project_root, "dbt_project.yml")
+    return yaml.safe_load(file_yaml)
+
+
+def set_project_config(project, config):
+    config_yaml = yaml.safe_dump(config)
+    write_file(config_yaml, project.project_root, "dbt_project.yml")
+
+
+def get_model_file(project, model: str) -> str:
+    return read_file(project.project_root, "models", f"{model}.sql")
+
+
+def set_model_file(project, model: str, model_sql: str):
+    write_file(model_sql, project.project_root, "models", f"{model}.sql")
+
+
+def assert_proper_scenario(
+    on_configuration_change,
+    results,
+    logs,
+    status: RunStatus,
+    messages_in_logs: List[str] = None,
+    messages_not_in_logs: List[str] = None,
+):
+    assert len(results.results) == 1
+    result = results.results[0]
+
+    assert result.node.config.on_configuration_change == on_configuration_change
+    assert result.status == status
+    for message in messages_in_logs or []:
+        assert_message_in_logs(logs, message)
+    for message in messages_not_in_logs or []:
+        assert_message_in_logs(logs, message, expected_fail=True)
```

Comment:
discussed async - we're primarily testing that the expected scenarios occurred by inspecting logs (dbt logs) which can be brittle if we want to change/remove logs or error-prone if we change functionality but keep logs as-is. Let's see if we can instead assert the expected sql statements were sent to the db for execution.

Reply:
Agreed, it's on the list of follow up items to take care of ahead of the full release.